proxy/server.py

```diff
@@ -44,768 +44,1020 @@
-    httptype = "HTTP"
-else:
-    # run in http mode
-    cookiesuffix = "path=/;"
-    httptype = "HTTP"
-
-# Logging
+DISABLED: Final[Set[str]] = set([
+    '/api/customer/registration',
+])
+WEB_ROOT: Final[str] = os.path.join(os.path.dirname(__file__), "web")
+SERVER_DEBUG: Final[bool] = bool(os.getenv("PW_DEBUG", "no").lower() == "yes")
+
+# Logging configuration
 log = logging.getLogger("proxy")
 logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.INFO)
-log.setLevel(logging.INFO)
-
-if debugmode:
-    log.info("pyPowerwall [%s] Proxy Server [%s] - %s Port %d - DEBUG" %
-             (pypowerwall.version, BUILD, httptype, port))
-    pypowerwall.set_debug(True)
-    log.setLevel(logging.DEBUG)
-else:
-    log.info("pyPowerwall [%s] Proxy Server [%s] - %s Port %d" %
-             (pypowerwall.version, BUILD, httptype, port))
-log.info("pyPowerwall Proxy Started")
-
-# Check for cache expire time limit below 5s
-if cache_expire < 5:
-    log.warning("Cache expiration set below 5s (PW_CACHE_EXPIRE=%d)" % cache_expire)
+log.setLevel(logging.DEBUG if SERVER_DEBUG else logging.INFO)
 
 # Signal handler - Exit on SIGTERM
 # noinspection PyUnusedLocal
 def sig_term_handle(signum, frame):
     raise SystemExit
 
-# Register signal handler
 signal.signal(signal.SIGTERM, sig_term_handle)
 
+if SERVER_DEBUG:
+    pypowerwall.set_debug(True)
+
+
+class CONFIG_TYPE(StrEnum):
+    """_summary_
+
+    Args:
+        StrEnum (_type_): _description_
+    """
+    PW_AUTH_MODE = auto()
+    PW_AUTH_PATH = auto()
+    PW_BIND_ADDRESS = auto()
+    PW_BROWSER_CACHE = auto()
+    PW_CACHE_EXPIRE = auto()
+    PW_CACHE_FILE = auto()
+    PW_CONTROL_SECRET = auto()
+    PW_COOKIE_SUFFIX = auto()
+    PW_EMAIL = auto()
+    PW_GW_PWD = auto()
+    PW_HOST = auto()
+    PW_HTTP_TYPE = auto()
+    PW_HTTPS = auto()
+    PW_NEG_SOLAR = auto()
+    PW_PASSWORD = auto()
+    PW_POOL_MAXSIZE = auto()
+    PW_PORT = auto()
+    PW_SITEID = auto()
+    PW_STYLE = auto()
+    PW_TIMEOUT = auto()
+    PW_TIMEZONE = auto()
+
+
+class PROXY_STATS_TYPE(StrEnum):
+    """_summary_
+
+    Args:
+        StrEnum (_type_): _description_
+    """
+    AUTH_MODE = auto()
+    CF = auto()
+    CLEAR = auto()
+    CLOUDMODE = auto()
+    CONFIG = auto()
+    COUNTER = auto()
+    ERRORS = auto()
+    FLEETAPI = auto()
+    GETS = auto()
+    MEM = auto()
+    MODE = auto()
+    POSTS = auto()
+    PW3 = auto()
+    PYPOWERWALL = auto()
+    SITE_NAME = auto()
+    SITEID = auto()
+    START = auto()
+    TEDAPI = auto()
+    TEDAPI_MODE = auto()
+    TIMEOUT = auto()
+    TS = auto()
+    UPTIME = auto()
+    URI = auto()
+
+# Configuration for Proxy - Check for environmental variables
+#    and always use those if available (required for Docker)
+# Configuration - Environment variables
+type PROXY_CONFIG = Dict[CONFIG_TYPE, str | int | bool | None]
+type PROXY_STATS = Dict[PROXY_STATS_TYPE, str | int | bool | None | PROXY_CONFIG | Dict[str, int]]
```

Comment:
The `type` constructs make this Python 3.12+ only. How difficult would it be to make it backward compatible for Python 3? I can update the library to be >= 3.12 but that seems like a big lift. Thoughts?

Reply:
I could, but I think if we do that we should move this to a class. Look at how unwieldy the type is... A dataclass type would be better and still backwards compatible.